manager/kitchen_manager.py

```diff
@@ -0,0 +1,202 @@
+    """
+    Write in file function name, arguments and their values
+    """
+
+    def wrapper(self, *args, **kwargs):
+        with open("results.txt", "w") as file:
+            parameter_names = list(inspect.signature(func).parameters.keys())
+            for index, argument in enumerate(args, 1):
+                file.write(f"{func.__name__}:{parameter_names[index]}={argument}\n")
+            for key, value in kwargs.items():
+                file.write(f"{func.__name__}: {key}={value}\n")
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+def limit_of_cals(func):
+    def wrapper(*args, **kwargs):
+        if wrapper.counter <= 3:
+            wrapper.counter += 1
+            return func(*args, **kwargs)
+        else:
+            raise Exception(f"Method '{func.__name__}' has reached the call limit of 3")
+
+    wrapper.counter = 0
+    return wrapper
+
+
+class KitchenManager:
+    """
+    Class creates an array of objects and helps to call main function
+    :param kitchens - an array of abstract class`s children objects
+    """
+    kitchens = []
+
+    @write_attribs_in_file
+    def add_kitchen(self, kitchen):
+        """
+        Method which adds kitchens to array
+        :param kitchen: name of objects in general
+        :return: an addition of kitchen to array
+        """
+        self.kitchens.append(kitchen)
+
+    @write_attribs_in_file
+    def find_bigger_than(self, size):
+        """
+        Method finds all objects bigger than given size
+        :param size: of place
+        :return: a filtered list with needed objects
+        """
+        return list(filter(lambda kitchen: kitchen.size > size, self.kitchens))
+
+    @write_attribs_in_file
+    def find_bigger_capacity_than(self, capacity):
+        """
+        Method finds all objects with bigger capacity than given size
+        :param capacity: of possible people in place
+        :return: a filtered list with needed objects
+        """
+        return list(filter(lambda kitchen: kitchen.capacity > capacity, self.kitchens))
+
+    def __len__(self):
+        """
+        Returns the number of kitchens in the manager
+        """
+        return len(self.kitchens)
+
+    def __getitem__(self, index):
+        """
+        Returns the kitchen at the specified index
+        """
+        return self.kitchens[index]
+
+    def __iter__(self):
+        """
+        Returns an iterator over the kitchens in the manager
+        """
+        return iter(self.kitchens)
+
+    @limit_of_cals
+    def type_of_kitchens(self):
+        """
+        :return: Type of kitchen for every kitchen
+        """
+        return [kitchen.kitchen_type() for kitchen in self.kitchens]  # TODO check kitchen_type(), why random?
+
+    @limit_of_cals
+    def enumerate_concatenation(self):
+        """
+        Returns a concatenation of each kitchen object and its index in the manager
+        """
+        return list(enumerate(self.kitchens))
```

Comment:
no need to cast it to list immideately. If you want it as list, you can cast it right there. In most cases, it would be more useful as simple iterator